### DISCUSSION TITLE
Have "rust: cargo build" task work on global level (Not just for workspace)

### DISCUSSION THREAD

**Discussion by trymeouteh** [rust-lang/rust-analyzer#18425] [Ideas]
I was able to create this debug launch config for my workspace by adding this config in `.vscode/launch.json`

```
{
	"name": "Rust",
	"type": "lldb",
	"request": "launch",
	"program": "${workspaceFolder}/target/debug/${workspaceFolderBasename}",
	"preLaunchTask": "rust: cargo build"
}
```

And it works!

However I want to use the same debug launch config everywhere. I added this config in the global `settings.json` file and when I run the *Rust* debug config, I get the following popup error

```
Could not find the task 'rust: cargo build'
```

Please add this task and perhaps other tasks to allow this simple debug config to work globally.

**Comment by attilarepka** [comment]
Hello @trymeouteh,
what you want is achievable by modifying `settings.json` (`Ctrl  + Shift + P` -> **Open User Settings (JSON)**).

```json
{
    ... // rest of your configuration
    ]
    "launch": {
        "version": "0.2.0",
        "configurations": [
            {
                "name": "Rust",
                "type": "lldb",
                "request": "launch",
                "program": "${workspaceFolder}/target/debug/${workspaceFolderBasename}",
                "preLaunchTask": "cargo build"
            }
        ]
    }
}
```
Notice that I modified `"rust: cargo build"` to `"cargo build"`

After this all you have to do is add a global task using `Ctrl + Shift + P` -> **Tasks: Open User Tasks**.
This should open up `tasks.json` file (or if not existing choose Others), where you can extend you existing tasks with:
```json
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "cargo build",
            "type": "shell",
            "command": "cargo build",
            "problemMatcher": [
                "$rustc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },            
        }
    ]
}
```

After this, you can run your global Debug builds in every workspace.

Hope this helps.
